### PR TITLE
feat(s3stream): change append callback thread number

### PR DESCRIPTION
**Pull request by lifepuzzlefun**

### CODE REVIEW

**Comment by lifepuzzlefun** [comment]
when single node in high produce qps the thread may not enough cause high produce latency.
image

**Comment by superhx** [comment]
> when single node in high produce qps ( 8w qps for 8c16g single node ) the thread may not enough cause high produce latency. image

Could we automatically make them scaling by CPU_CORES likes
```
    public StreamReaders(long size, ObjectManager objectManager, ObjectStorage objectStorage,
        ObjectReaderFactory objectReaderFactory) {
        this(size, objectManager, objectStorage, objectReaderFactory, Systems.CPU_CORES);
    }
```
Currently, the thread pool threads number is based on 2 Core

**Comment by lifepuzzlefun** [comment]
> > when single node in high produce qps ( 8w qps for 8c16g single node ) the thread may not enough cause high produce latency. image
> 
> Could we automatically make them scaling by CPU_CORES likes
> 
> ```
>     public StreamReaders(long size, ObjectManager objectManager, ObjectStorage objectStorage,
>         ObjectReaderFactory objectReaderFactory) {
>         this(size, objectManager, objectStorage, objectReaderFactory, Systems.CPU_CORES);
>     }
> ```
> 
> Currently, the thread pool threads number is based on 2 Core.

agree. and do you think we should introduce the thread number or just automatically set to 2 * Systems.CPU_CROES.

for 2cpu deploy the thread number will be drop from 8 to 4.

or keep the config and change the default to 2 * CPU

**Comment by superhx** [comment]
> > > when single node in high produce qps ( 8w qps for 8c16g single node ) the thread may not enough cause high produce latency. image
> > 
> > 
> > Could we automatically make them scaling by CPU_CORES likes
> > ```
> >     public StreamReaders(long size, ObjectManager objectManager, ObjectStorage objectStorage,
> >         ObjectReaderFactory objectReaderFactory) {
> >         this(size, objectManager, objectStorage, objectReaderFactory, Systems.CPU_CORES);
> >     }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Currently, the thread pool threads number is based on 2 Core.
> 
> agree. and do you think we should introduce the thread number or just automatically set to 2 * Systems.CPU_CROES.
> 
> for 2cpu deploy the thread number will be drop from 8 to 4.
> 
> or keep the config and change the default to 2 * CPU

I think auto-set is enough
- APPEND_CALLBACK_EXECUTOR could be Systems.CPU_CROES * 2: it just send response the channel
- READ_ASYNC_EXECUTOR could be Systems.CPU_CROES * 4：it has blocking operations.

File: s3stream/src/main/java/com/automq/stream/utils/threads/S3StreamThreadPoolMonitor.java [hunk 1]
```diff
@@ -38,6 +38,11 @@ public static void config(Logger waterMarkLoggerConfig, long threadPoolStatusPer
         threadPoolStatusPeriodTime = threadPoolStatusPeriodTimeConfig;
     }
 
+    public static ThreadPoolExecutor newSingleThreadExecutor(String name, ThreadFactory factory) {
```
Comment (superhx) [inline s3stream/src/main/java/com/automq/stream/utils/threads/S3StreamThreadPoolMonitor.java:41]: You could use Threads#newFixedThreadPoolWithMonitor

File: core/src/main/scala/kafka/log/streamaspect/ElasticLog.scala [hunk 2]
```diff
@@ -588,14 +589,16 @@ object ElasticLog extends Logging {
...
     private val APPEND_ACK_TIME_HIST = KafkaMetricsGroup.newHistogram("AppendAckTimeNanos")
-    private val APPEND_CALLBACK_EXECUTOR: Array[ExecutorService] = new Array[ExecutorService](8)
-    private val READ_ASYNC_EXECUTOR: Array[ExecutorService] = new Array[ExecutorService](8)
+    private val APPEND_CALLBACK_EXECUTOR: Array[ExecutorService] = new Array[ExecutorService](Systems.CPU_CORES * 2)
+    private val READ_ASYNC_EXECUTOR: Array[ExecutorService] = new Array[ExecutorService](Systems.CPU_CORES * 4)
 
     for (i <- APPEND_CALLBACK_EXECUTOR.indices) {
-        APPEND_CALLBACK_EXECUTOR(i) = Executors.newSingleThreadExecutor(ThreadUtils.createThreadFactory("log-append-callback-executor-" + i, true))
+        val name = "log-append-callback-executor-" + i
+        APPEND_CALLBACK_EXECUTOR(i) = S3StreamThreadPoolMonitor.newSingleThreadExecutor(name, ThreadUtils.createThreadFactory(name, true))
```
Comment (superhx) [inline core/src/main/scala/kafka/log/streamaspect/ElasticLog.scala:597]: You could use Threads#newFixedThreadPoolWithMonitor

**Review by superhx — APPROVED** [review]

**Pull request merged**